packages/jsii-pacmak/lib/targets/java.ts

```diff
@@ -869,30 +869,47 @@ class JavaGenerator extends Generator {
       nullable: !!property.optional,
       fieldName: this.code.toCamelCase(safeName),
       fieldJavaType: this.toJavaType(property.type),
-      fieldJavaClass: `${this.toJavaType(property.type, true)}.class`,
+      fieldJavaClass: `${this.toJavaType(property.type, true).syntax}.class`,
       javaTypes: this.toJavaTypes(property.type),
       immutable: property.immutable || false,
       inherited,
     };
   }
 
   private emitBuilderSetter(prop: JavaProp, builderName: string) {
+    const summary = (prop.docs && prop.docs.summary) || 'the value to be set';
     for (const type of prop.javaTypes) {
       this.code.line();
       this.code.line('/**');
       this.code.line(` * Sets the value of ${prop.propName}`);
-      const summary = (prop.docs && prop.docs.summary) || 'the value to be set';
       this.code.line(` * ${paramJavadoc(prop.fieldName, prop.nullable, summary)}`);
       this.code.line(' * @return {@code this}');
       if (prop.docs && prop.docs.deprecated) {
         this.code.line(` * @deprecated ${prop.docs.deprecated}`);
       }
       this.code.line(' */');
       this.emitStabilityAnnotations(prop.spec);
-      this.code.openBlock(`public ${builderName} ${prop.fieldName}(${type} ${prop.fieldName})`);
+      this.code.openBlock(`public ${builderName} ${prop.fieldName}(${type.syntax} ${prop.fieldName})`);
       this.code.line(`this.${prop.fieldName} = ${prop.fieldName};`);
       this.code.line('return this;');
       this.code.closeBlock();
+
+      if (type.listOf && type.listOf.syntax !== 'java.lang.Object') {
```

Comment:
This is just to catch types into which `java.util.List` could be assigned, right? I would make that a function to be super clear about that:

```
if (type.listOf && !listTypeAssignableTo(type.listOf)) { 
  ...
}
```

But is this even a good idea? Don't you think it would be confusing that you can call:

```java
builder.someStrings("a", "b", "c");
```

But will not be able to call:

```java
builder.someAnything(a, b, c); // Oops won't work! Why not? I've been trained that I can do that!
```

We do have a number of cases in the CDK where we use `any`...